src/core.ts

```diff
@@ -92,10 +92,18 @@ export function addEvent(type: string, eventState: any, time?: number) {
   }
   nextPayload.push(eventStr);
   nextPayloadLength += eventStr.length;
-  if (timeout) {
-    clearTimeout(timeout);
+
+  // Edge case:
+  // Don't reschedule upload when next payload consists of exactly one XhrError instrumentation event.
+  // This helps us avoid the infinite loop in the case when all requests fail (e.g. dropped internet connection)
```

Comment:
This is not for the retries, but rather for sending new payload. When some xhr fails, we store its payload for retry, but we generate a new instrumentation event, which becomes a new payload of its own. Then, this new payload consisting of just this event is scheduled to be sent through a timeout, and, if it fails to deliver as well, we get into a loop.